bindgen/ir/var.rs

```diff
@@ -389,9 +392,94 @@ impl ClangSubItemParser for Var {
     }
 }
 
+fn parse_macro_clang_fallback(
+    ctx: &mut BindgenContext,
+    cursor: &clang::Cursor,
+) -> Option<(Vec<u8>, cexpr::expr::EvalResult)> {
+    let file = ".macro_eval.c".to_string();
+    let contents = format!("int main() {{ {}; }}", cursor.spelling(),);
+
+    let root = if let Some(ftu) = ctx.fallback_translation_unit_mut() {
+        ftu.reparse(&[clang::UnsavedFile::new(&file, &contents)]);
+        ftu.translation_unit().cursor().collect_children()
+    } else {
+        let index = clang::Index::new(false, false);
+
+        let mut c_args = Vec::new();
+        for input_header in ctx.options().input_headers.iter() {
```

Comment:
Would be nice to change this to something like:

```rust
let ftu = ctx.ensure_fallback_translation_unit()?;
ftu.reparse(&[clang::UnsavedFile::new(&file, &contents)]);
ftu.translation_unit().cursor().collect_children()
```

Where `ensure_fallback_translation_unit` does the hard part of building up the translation unit perhaps?

Reply:
Yeah, assuming that wouldn't meaningfully affect performance that'd be nice.